### ISSUE TITLE
[Data] Small Dataset Size - Only 75/460 Utterances Segmented

### ISSUE THREAD

**Issue by faransansj** [faransansj/Project_Sullivan#3]
## Issue

Currently using only **75 out of 460** available utterances from USC-TIMIT dataset (16% utilization).

### Dataset Split
```json
{
  "total_utterances": 75,
  "total_subjects": 15,
  "splits": {
    "train": 50 utterances (10 subjects),
    "val": 10 utterances (2 subjects),
    "test": 15 utterances (3 subjects)
  }
}
```

### Available Data
- **Segmented**: 75 utterances (186,124 frames)
- **Downloaded but not segmented**: 393 utterances
- **Total available**: 468 utterances in USC-TIMIT dataset
- **Utilization**: 16%

### Data Scarcity Issues
1. **Small training set**: Only 50 utterances (10 subjects)
2. **Overfitting risk**: High (observed 27% train/val gap)
3. **Generalization**: Poor due to limited speaker diversity
4. **Performance ceiling**: Dataset size limits achievable accuracy

### Evidence of Data Hunger
From baseline training:
- Training loss: 0.665 (final)
- Validation loss: 0.9124 (best)
- **Overfitting gap**: 27%
- **Conclusion**: Model memorizes rather than learns patterns

### Segmentation Time Constraint
- **U-Net segmentation**: ~22.8 fps (frames per second)
- **Current 75 utterances**: 2.3 hours of processing
- **All 460 utterances**: **~14 hours** estimated
- **Decision**: Started with subset for rapid prototyping

### Resource Limitations
- Segmentation requires GPU or long CPU time
- Phase 1 focused on pipeline validation
- Prioritized getting to Phase 2 quickly

#### Approach 1: Local GPU Segmentation (If GPU fixed)
```bash
python scripts/segment_full_dataset.py --batch-size 4 --device cuda
```
- **Time**: ~14 hours (estimated)
- **Requires**: GPU issue #1 resolved OR PyTorch downgrade
- **Output**: ~1.14M additional frames

#### Approach 2: CPU Segmentation (Overnight)
```bash
python scripts/segment_full_dataset.py --batch-size 2 --device cpu
```
- **Time**: ~30-40 hours (CPU slower)
- **Strategy**: Run overnight + weekends
- **No GPU dependency**

#### Approach 3: Colab Segmentation
- Use free Colab GPU for batch segmentation
- Download MRI data to Colab
- Run segmentation in chunks
- Download results

### Expected Improvements

With 460 utterances (6× more data):
- **Reduced overfitting**: More diverse training examples
- **Better generalization**: More speakers, more variability
- **Improved metrics**: Estimated 20-30% RMSE reduction
- **More robust model**: Better handling of unseen speakers

## Data Splits After Full Segmentation

Proposed split (70/15/15):
- **Train**: 322 utterances (~30 subjects)
- **Val**: 69 utterances (~7 subjects)
- **Test**: 69 utterances (~7 subjects)
- **Total**: 460 utterances (~45 subjects)

**6× increase** in training data

### High Priority Because:
1. **Data is the bottleneck**: More data = better models
2. **Already downloaded**: No additional download needed
3. **Proven pipeline**: Segmentation works reliably
4. **High ROI**: ~14 hours → 6× more data

### Not Critical Because:
1. **Can start with 75**: Already training models
2. **Phase 2-B first**: Test Transformer on current data
3. **GPU blocked**: Limited compute anyway (see #1)
4. **Time availability**: Can run overnight/weekends

### Phase A: Validate on Current Data (Current)
- [x] Train baseline on 75 utterances
- [ ] Train Transformer on 75 utterances
- [ ] Evaluate if data is bottleneck or architecture

### Phase B: Segment Additional Data (After GPU fix or in parallel)
- [ ] Set up segmentation job (CPU or GPU)
- [ ] Segment remaining 385 utterances
- [ ] Extract articulatory parameters
- [ ] Create new train/val/test splits

### Phase C: Retrain with Full Dataset
- [ ] Retrain best model with 6× more data
- [ ] Compare performance improvement
- [ ] Validate generalization on larger test set

## Blockers

- ⚠️ GPU access (#1) - affects segmentation speed
- ✅ Segmentation pipeline - working and validated
- ✅ Parameter extraction - working and validated
- ✅ Data download - complete

## Related Issues

- #1 - GPU compatibility (affects segmentation speed)
- #2 - Performance issues (data expansion may help)

## Related Files

- Segmentation script: `scripts/segment_full_dataset.py`
- Dataset stats: `data/processed/splits/split_summary.json`
- Segmentation report: `docs/PROJECT_SULLIVAN_SEGMENTATION_COMPLETE.md`

## Success Criteria

- [ ] All 460 utterances segmented
- [ ] New train/val/test splits created
- [ ] Models retrained with full dataset
- [ ] Performance improvement documented

---

**Priority**: 🟡 MEDIUM (High ROI but not blocking)  
**Milestone**: M2 optimization
**Type**: Data expansion
